checker/check_test.go

```diff
@@ -0,0 +1,270 @@
+	g.Expect(status.Code).To(Equal(OK))
+
+	http := req.GetAttributes().GetRequest().GetHttp()
+	http.Method = "HEAD"
+
+	status = checkStore(store, req)
+	g.Expect(status.Code).To(Equal(PERMISSION_DENIED))
+}
+
+// And endpoint with no Policies should evaluate Profiles.
+func TestCheckStoreProfileOnly(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	store := policystore.NewPolicyStore()
+	store.Endpoint = &proto.WorkloadEndpoint{
+		Tiers:      []*proto.TierInfo{{}},
+		ProfileIds: []string{"profile1", "profile2"},
+	}
+	store.ProfileByID[proto.ProfileID{Name: "profile1"}] = &proto.Profile{
+		InboundRules: []*proto.Rule{
+			{
+				Action: "Deny",
+				Http:   &proto.HTTPSelector{Methods: []string{"HEAD"}},
+			},
+		},
+	}
+	store.ProfileByID[proto.ProfileID{Name: "profile2"}] = &proto.Profile{
+		InboundRules: []*proto.Rule{
+			{
+				Action: "allow",
+				Http:   &proto.HTTPSelector{Methods: []string{"GET"}},
+			},
+		},
+	}
+
+	req := &authz.CheckRequest{Attributes: &authz.AttributeContext{
+		Source: &authz.AttributeContext_Peer{
+			Principal: "spiffe://cluster.local/ns/default/sa/steve",
+		},
+		Request: &authz.AttributeContext_Request{
+			Http: &authz.AttributeContext_HTTPRequest{Method: "GET"},
+		},
+	}}
+
+	status := checkStore(store, req)
+	g.Expect(status.Code).To(Equal(OK))
+
+	http := req.GetAttributes().GetRequest().GetHttp()
+	http.Method = "HEAD"
+
+	status = checkStore(store, req)
+	g.Expect(status.Code).To(Equal(PERMISSION_DENIED))
+}
+
+// Ensure policy action of "Pass" ends policy evaluation and moves to profiles.
+func TestCheckStorePass(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	store := policystore.NewPolicyStore()
+	store.Endpoint = &proto.WorkloadEndpoint{
+		Tiers: []*proto.TierInfo{{
+			Name:            "tier1",
+			IngressPolicies: []string{"policy1", "policy2"},
+		}},
+		ProfileIds: []string{"profile1"},
+	}
+
+	// Policy1 matches and has action PASS, which means policy2 is not evaluated.
+	store.PolicyByID[proto.PolicyID{Tier: "tier1", Name: "policy1"}] = &proto.Policy{
+		InboundRules: []*proto.Rule{
+			{
+				Action: "pass",
+				Http:   &proto.HTTPSelector{Methods: []string{"GET"}},
+			},
+		},
+	}
+	store.PolicyByID[proto.PolicyID{Tier: "tier1", Name: "policy2"}] = &proto.Policy{
+		InboundRules: []*proto.Rule{
+			{
+				Action: "deny",
+				Http:   &proto.HTTPSelector{Methods: []string{"GET"}},
+			},
+		},
+	}
+
+	// Profile1 matches and allows the traffic.
+	store.ProfileByID[proto.ProfileID{Name: "profile1"}] = &proto.Profile{
+		InboundRules: []*proto.Rule{
+			{
+				Action: "allow",
+				Http:   &proto.HTTPSelector{Methods: []string{"HEAD", "GET"}},
+			},
+		},
+	}
+
+	req := &authz.CheckRequest{Attributes: &authz.AttributeContext{
+		Source: &authz.AttributeContext_Peer{
+			Principal: "spiffe://cluster.local/ns/default/sa/steve",
+		},
+		Request: &authz.AttributeContext_Request{
+			Http: &authz.AttributeContext_HTTPRequest{Method: "GET"},
+		},
+	}}
+
+	status := checkStore(store, req)
```

Comment:
Is there a way in gomega to check if the call to `checkStore` invoked the api `checkProfile`?

Reply:
Not without making checkProfile a function pointer that I swap out.